atomix/utils/src/main/java/io/atomix/utils/net/Address.java

```diff
@@ -140,43 +140,20 @@ public InetAddress address() {
    * @return the IP address
    */
   public InetAddress address(final boolean resolve) {
-    if (resolve) {
-      socketAddress = resolveAddress();
+    if (resolve || socketAddress.isUnresolved()) {
+      // the constructor will by default attempt to resolve the host, and will fallback to the an
+      // unresolved address if it couldn't
+      socketAddress = new InetSocketAddress(host, port);
```

Comment:
That's my understanding - the boolean here refers to whether or not you should force the resolve even if it was already resolved. It will still fallback to the given host if it can't resolve anything. I checked usages, and this is called when bootstrapping client connections (where we force resolve, which makes sense), and in the V1 of the messaging protocol - but I think we always use V2 now, which encodes the host, and doesn't call this method.